lib/core/client.js

```diff
@@ -443,7 +443,13 @@ class NightwatchClient extends EventEmitter {
   }
 
   setLaunchUrl() {
-    const value = this.settings.baseUrl || this.settings.launchUrl || this.settings.launch_url || null;
+    let value = this.settings.baseUrl || this.settings.launchUrl || this.settings.launch_url || null;
+
+    // For e2e and component testing on android emulator
+    if (this.settings.desiredCapabilities.avd) {         
```

Comment:
Add a check of `real_mobile` capability to identify the android emulator environment with `avd`.
```suggestion
    if (!this.settings.desiredCapabilities.real_mobile && this.settings.desiredCapabilities.avd) {         
```